prog.go

```diff
@@ -565,37 +622,44 @@ var haveProgTestRun = internal.FeatureTest("BPF_PROG_TEST_RUN", "4.12", func() e
 	return err
 })
 
-func (p *Program) testRun(in []byte, repeat int, reset func()) (uint32, []byte, time.Duration, error) {
-	if uint(repeat) > math.MaxUint32 {
-		return 0, nil, 0, fmt.Errorf("repeat is too high")
+func (p *Program) testRun(opts *RunOptions) (uint32, time.Duration, error) {
+	if len(opts.Data) == 0 {
+		return 0, 0, fmt.Errorf("missing input")
 	}
 
-	if len(in) == 0 {
-		return 0, nil, 0, fmt.Errorf("missing input")
+	if uint(len(opts.Data)) > math.MaxUint32 {
+		return 0, 0, fmt.Errorf("input is too long")
 	}
 
-	if uint(len(in)) > math.MaxUint32 {
-		return 0, nil, 0, fmt.Errorf("input is too long")
+	if err := haveProgTestRun(); err != nil {
+		return 0, 0, err
 	}
 
-	if err := haveProgTestRun(); err != nil {
-		return 0, nil, 0, err
+	ctx := new(bytes.Buffer)
```

Comment:
Nit: this forces an allocation even if we don't have opts.Context. We can fix that in a follow up though.

Reply:
Ok, will send followup